[SPARK-48]  Unit Tests for mechaclient.py

shatteredbeam opened the pull request:
Unit tests for mechaclient.py

shatteredbeam committed:
add fixtures to conftest.py, new file created for mechaclient tests and added mark to pytest.ini

shatteredbeam committed:
Add tests for mechaclient.py

theunkn0wn1 commented on tests/conftest.py:
```diff
@@ -315,4 +316,4 @@ def test_fact_fx() -> Fact:
                 mfd=False,
                 edited=None
-                )
+                )
```
missing \n at EOL

shatteredbeam committed:
add missing newline.

theunkn0wn1 commented on tests/test_mechaclient.py:
```diff
@@ -0,0 +1,168 @@
+    Asserts the version is set, and returning string.
+    """
+    assert bot_fx.__version__
+    assert isinstance(bot_fx.__version__, str)
```
[scope] testing builtin.
[error] testing behavior of hardcoded attribute.

shatteredbeam replied:
Superfluous test removed in 96dac7b.

Eearslya commented on tests/conftest.py:
```diff
@@ -37,6 +37,7 @@
 
 setup_logging("logs/unit_tests.log")
 
+from src.mechaclient import MechaClient
```
[Style] (unused-import) What does adding this import provide? I can't see any additional code that uses this import.

theunkn0wn1 replied:
Id argue this is convention / refactor more than it is style, but yes, this appears to be a needless import.

shatteredbeam replied:
If its an invalid import, I'll remove it.  I'll double check.

shatteredbeam replied:
unused import removed in 96dac7b.

Eearslya replied:
I believe you removed the wrong import; the one I was referring to is in `conftest.py`, which is still there.

theunkn0wn1 replied:
This resolved?

Eearslya commented on tests/test_mechaclient.py:
```diff
@@ -0,0 +1,168 @@
+            ...
+
+    bot_fx.fact_manager = FakeFactManager()
+    assert isinstance(bot_fx._fact_manager, FakeFactManager)
```
[Discussion] Should these tests be separated? Given the parametrize, it's testing both a good path and a bad path with each invocation.

Eearslya commented on tests/test_mechaclient.py:
```diff
@@ -0,0 +1,168 @@
+    bot_fx.board = RatBoard()
+
+    assert isinstance(bot_fx._rat_board, RatBoard)
+    assert bot_fx.board is bot_fx._rat_board
```
[Discussion] Should these tests be separated? Given the parametrize, it's testing both a good path and a bad path with each invocation.

shatteredbeam replied:
They could be separated, but what would be gained from splitting it?  The tests do pass with known good or bad data.

Eearslya replied:
It's more of a convention thing IMO, making it so one test tests one thing. No real tangible benefit other than organization. (Which is still nice to have.)

theunkn0wn1 replied:
Arguably, tests should only test one scenario

Either test the good path, or the bad path, but not both within a test definition.

theunkn0wn1 commented on pytest.ini:
```diff
@@ -23,6 +23,7 @@ markers =
     ratsignal_parse: marks a test as a ratsignal parser test
     autocorrect: tests for the autocorrect util
     offline: offline-aware tests
+    mechaclient: marks a test as a mechaclient test.
```
```suggestion
    mechaclient: mechaclient tests
```

shatteredbeam replied:
Changed to 'mechaclient tests' in 96dac7b.

theunkn0wn1 commented on tests/test_mechaclient.py:
```diff
@@ -0,0 +1,168 @@
+
+def test_verify_api_handler(bot_fx):
+    """
+    Asserts the api_handler property returns an object, and cannot be set (no setter)
```
[error] this is not what the test is doing.

shatteredbeam replied:
Docstring fixed in 96dac7b.

theunkn0wn1 commented on tests/test_mechaclient.py:
```diff
@@ -0,0 +1,168 @@
+    class FakeAPIHandler:
+        ...
+
+    assert bot_fx.api_handler is None
```
[error] this is already tested in a different test case.

shatteredbeam replied:
Not specifically this one, removing the test removes coverage for the api_handler getter.

shatteredbeam committed:
[Spark-48] Reviews fix for PR #118:
[Spark-48] Removed unused import line
[Spark-48] Removed superfluous tests
[Spark-48] Fixed docstring for test_verify_api_handler

shatteredbeam committed:
[SPARK-48] Remove unused import from conftest.py
[SPARK-48] Split Fact Manager Deleter test into its own def
[SPARK-48] Split RatBoard Deleter test into own def.

theunkn0wn1 reviewed (approved):
LGTM
